dev/tests/api-functional/testsuite/Magento/GraphQl/UrlRewrite/UrlResolverTest.php

```diff
@@ -355,4 +356,42 @@ public function testResolveSlash()
         $this->assertEquals($targetPath, $response['urlResolver']['relative_url']);
         $this->assertEquals('CMS_PAGE', $response['urlResolver']['type']);
     }
+
+    /**
+     * Test for custom type which point to the valid product/category/cms page.
+     *
+     * @magentoApiDataFixture Magento/CatalogUrlRewrite/_files/product_with_category.php
+     */
+    public function testGetNonExistentUrlRewrite()
```

Comment:
In the scope of tests, we need to cover the case when `No such entity found with matching URL key` will get.

Reply:
Hi @naydav I've added a new Test case `testNonExistentEntityUrlRewrite` to cover `No such entity found with matching URL key`. Please take a look into it.